src/Microsoft.AspNetCore.Sockets.Client.Http/Internal/TaskExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Sockets.Http.Internal
+{
+    public static class NoThrowTaskExtensions
```

Comment:
the class should be called TaskExtensions. We may want to add things (like OrTimeout()) here in the future - you could say `_eventQueue.Drain().NoThrow().OrTimeout()` (the order is important).